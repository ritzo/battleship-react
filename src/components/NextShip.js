import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './NextShip.css';

// eslint-disable-next-line react/prefer-stateless-function
class NextShip extends React.Component {
  render() {
    const { availableShips } = this.props;
    const [nextShip] = availableShips;

    return (
      <div>
        <p>
          <span>Next ship: </span>
          { nextShip }
        </p>

        <p>
          <span>Ships left: </span>
          { availableShips.slice(1).join(', ') }
        </p>
      </div>
    );
  }
}

NextShip.propTypes = {
  availableShips: PropTypes.arrayOf(PropTypes.number).isRequired,
};

function mapStateToProps(state) {
  return {
    availableShips: state.availableShips,
  };
}

export default connect(mapStateToProps)(NextShip);
