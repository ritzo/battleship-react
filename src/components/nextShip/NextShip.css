.next-ship-container {
    text-align: center;
}
.edition-hint {
    font-size: 0.8em;
    font-weight: bold;
}