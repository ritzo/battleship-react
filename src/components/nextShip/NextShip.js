import React from 'react';
import PropTypes from 'prop-types';

import './NextShip.css';

function NextShip(props) {
  const { availableShips } = props;
  const [nextShip] = availableShips;

  if (!nextShip) {
    return (
      <div>
        <p>The fleet is ready</p>
      </div>
    );
  }

  return (
    <div>
      <p>
        <span>Next ship: </span>
        { nextShip }
      </p>

      <p>
        <span>Ships left: </span>
        { availableShips.slice(1).join(', ') }
      </p>
    </div>
  );
}

NextShip.propTypes = {
  availableShips: PropTypes.arrayOf(PropTypes.number).isRequired,
};

export default NextShip;
