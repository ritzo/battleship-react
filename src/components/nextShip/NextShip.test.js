import React from 'react';
import { shallow } from 'enzyme';

import NextShip from './NextShip';

describe('NextShip', () => {
  it('Snapshot', () => {
    const availableShips = [2, 3];

    const enzymeWrapper = shallow(
      <NextShip availableShips={availableShips} />,
    );

    expect(enzymeWrapper).toMatchSnapshot();
  });
});
