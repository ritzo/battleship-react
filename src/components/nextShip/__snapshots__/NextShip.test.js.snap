// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextShip Snapshot 1`] = `
<div>
  <p>
    <span>
      Next ship: 
    </span>
    2
  </p>
  <p>
    <span>
      Ships left: 
    </span>
    3
  </p>
</div>
`;
